File: frontend/src/types/Book.ts
// Book interface - matches the backend DTO
export interface Book {
  id: number;
  title: string;
  author: string;
  description?: string;
  createdAt: string;
  updatedAt?: string;
}

// DTO for creating a new book (no id needed)
export interface CreateBookDto {
  title: string;
  author: string;
  description?: string;
}

// DTO for updating a book
export interface UpdateBookDto {
  title: string;
  author: string;
  description?: string;
}

File: backend/LibraryAPI/Models/Book.cs
using System.ComponentModel.DataAnnotations;

namespace LibraryAPI.Models
{
    public class Book
    {
        [Key]
        public int Id { get; set; }

        // Book Details
        
        [Required(ErrorMessage = "Title is required")]
        [StringLength(200, ErrorMessage = "Title cannot exceed 200 characters")]
        public string Title { get; set; } = string.Empty;

        [Required(ErrorMessage = "Author is required")]
        [StringLength(100, ErrorMessage = "Author cannot exceed 100 characters")]
        public string Author { get; set; } = string.Empty;

        [StringLength(1000, ErrorMessage = "Description cannot exceed 1000 characters")]
        public string? Description { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime? UpdatedAt { get; set; }
    }
}